frontend/apps/reader/modules/readertypography.lua

```diff
@@ -0,0 +1,802 @@
+                    end
+                end
+            end
+            -- Text might be too long for InfoMessage
+            local status_text = table.concat(lang_infos, "\n")
+            local TextViewer = require("ui/widget/textviewer")
+            local Font = require("ui/font")
+            UIManager:show(TextViewer:new{
+                title = _("Language tags (and hyphenation dictionaries) used since start up"),
+                text = status_text,
+                text_face = Font:getFace("smallinfont"),
+                height = Screen:getHeight() * 4/5,
+            })
+        end,
+        keep_menu_open = true,
+        separator = true,
+    })
+
+    for __, v in ipairs(LANGUAGES) do
+        local lang_tag, lang_aliases, lang_features, lang_name = unpack(v) -- luacheck: no unused
+
+        table.insert(self.language_submenu, {
+            text_func = function()
+                local text = lang_name
+                local feat = ""
+                if lang_features:find("H") then
+                    feat = feat .. "‐ "
+                end
+                if lang_features:find("b") then
+                    feat = feat .. ""
+                elseif lang_features:find("B") then
+                    feat = feat .. ""
+                end
+                -- Other usable nerdfont glyphs in case of need:
+                -- feat = feat .. "       "
+                text = text .. "   " .. feat
+                if lang_tag == G_reader_settings:readSetting("text_lang_default") then
+                    text = text .. "   ★"
+                end
+                if lang_tag == G_reader_settings:readSetting("text_lang_fallback") then
+                    text = text .. "   �"
+                end
+                return text
+            end,
+            callback = function()
+                UIManager:show(InfoMessage:new{
+                    text = T(_("Changed typography language to %1."), BD.wrap(lang_name)),
+                })
+                self.text_lang_tag = lang_tag
+                self.ui.document:setTextMainLang(lang_tag)
+                self.ui:handleEvent(Event:new("UpdatePos"))
+            end,
+            hold_callback = function(touchmenu_instance)
+                UIManager:show(MultiConfirmBox:new{
+                    -- No real need for a way to remove default one, we can just
+                    -- toggle between setting a default OR a fallback (if a default
+                    -- one is set, no fallback will ever be used - if a fallback one
+                    -- is set, no default is wanted; so when we set one below, we
+                    -- remove the other).
+                    text = T( _("Would you like %1 to be used as the default (★) or fallback (�) typography language?\n\nDefault will always take precedence while fallback will only be used if the language of the book can't be automatically determined."), BD.wrap(lang_name)),
```

Comment:
Hmm, that one's mildly trickier with my initial comment about typography/typography rules...

...) language for typography rules?

Maybe?

Reply:
I think language for typography rules is clearer than typography language.

Reply:
Thanks for the previous feedback.
One thing just to be sure: it's really typograph**y** rules/languages, and not typograph**ic** rules like I'd be inclined to write in french?

Reply:
I'd say they're both correct. Google Books seems to agree.

Reply:
Much like *règles de typographie* and *règles typographiques* in French :).